### PR TITLE
Remove trader repo if version does not match

### PR DESCRIPTION
**Pull request by jmoreira-valory**
Remove trader repo if current downloaded version does not match.

Addresses https://github.com/valory-xyz/trader-quickstart/issues/145

### CODE REVIEW

File: run_service.sh [hunk 1]
```diff
@@ -665,17 +665,27 @@ if [ "$rcp_new_filter_supported" = false ]; then
...
-    echo "Cloning the $directory repo from $org_name GitHub..."
+if [ -d "$directory" ]; then
+    current_version=$(git --git-dir="$directory/.git" describe --tags)
+
+    if [ "$current_version" != "$service_version" ]; then
+        echo "Current version of $directory ($current_version) does not match expected version ($service_version)."
+        echo "Removing '$directory' directory..."
+        echo ""
+        sudo rm -rf "$directory"
+    fi
```
Comment (Adamantios) [inline run_service.sh:683]: Maybe prompt the user to confirm that they do not mind losing any changes in the code.

Comment (jmoreira-valory) [reply run_service.sh:683]: I am unsure if this is required. @dagacha please confirm.

Comment (dagacha) [reply run_service.sh:683]: IMO it is not required. As an operator I need to actively download the new version in the repository for the code to change. So when I run the script I do it under the assumption that I want to change the code, so no further confirmation needed.

**Review by Adamantios — APPROVED** [review]

File: run_service.sh (hunk 1 above)
Comment (dagacha) [reply run_service.sh:683]: Confirmed to work on Ubuntu. This is whay you see:

`------------------------------
Setting up 'trader' repository
------------------------------

Current version of trader (v0.11.4) does not match expected version (v0.11.5).
Removing 'trader' directory...
`

Comment (dagacha) [reply run_service.sh:683]: Confirmed to work on MacOs 14.2

**Pull request merged**